src/ipc/ipc4/dai.c

```diff
@@ -31,6 +31,44 @@
 
 LOG_MODULE_DECLARE(ipc, CONFIG_SOF_LOG_LEVEL);
 
+int dai_set_link_hda_config(uint16_t *link_config,
+			struct ipc_config_dai *common_config,
+			const void *spec_config)
+{
```

Comment:
isn't this too HDA specific? I'd say it should go in the driver

Reply:
@lyakh @jxstelter This is a bit iffy but we do have DAI type specific code for all vendors here and we need to convert from the IPC structs into language of the Zephyr driver interface (e.g. a single link_config field).  At least in my initial review I checked that we need IPC4 structs the driver does not have access to.

@jxstelter Can you double-check could we calculate this in the driver? Direction and stream_id should be available already in driver code -- I guess depth and channel count should be available as well...?